src/docs/chunkcache.dox

```diff
@@ -1,4 +1,37 @@
-/*! @page chunkcache Chunkcache
+/*! @page chunkcache Chunk cache
+
+The chunk cache is a feature designed to work in conjunction with @ref
+arch-cloud-storage-extension. These extensions would typically need to do IO
+over the network, which comes with performance and API usage costs. To
+ameliorate these, WiredTiger can be configured to cache in-use content from
+these storage sources on a disk.
+
+Architecturally, the chunk cache sits between the block cache and the
+filesystem layer, caching data as it would appear "at rest". This implies
+the chunk cache stores data in its encrypted and compressed form, making it
+potentially more efficient. However, that also means it has no knowledge of
+the tree or page structure of the table, much like a file system wouldn't.
+
+The chunk cache is read-only, and will only cache read-only content. In
+practice, this means any tiered objects that have been "flushed" during a
+checkpoint and can no longer be altered. Any data that could still be
```

Comment:
Nitpick -> do we need `and` here?

Reply:
Yep - we're saying the content/tiered objects both (1) have been flushed during a checkpoint, and (2) can't be altered any more.